lib/semantics/check-io.h

```diff
@@ -117,28 +143,13 @@ class IoChecker : public virtual BaseChecker {
   void Init(IoStmtKind s) {
     stmt_ = s;
     specifierSet_.reset();
-    hasIoControlList_ = false;
-    hasInternalUnit_ = hasNumberUnit_ = hasStarUnit_ = false;
-    hasCharFormat_ = hasLabelFormat_ = hasStarFormat_ = false;
-    hasKnownAccess_ = hasAccessDirect_ = hasAccessStream_ = false;
-    hasAdvanceYes_ = false;
-    hasAsynchronousYes_ = false;
-    hasKnownStatus_ = hasStatusNew_ = hasStatusReplace_ = hasStatusScratch_ =
-        false;
-    hasDataList_ = false;
+    flag_.reset();
   }
 
   SemanticsContext &context_;
   IoStmtKind stmt_ = IoStmtKind::None;
   common::EnumSet<SpecifierKind, SpecifierKind_enumSize> specifierSet_;
-  bool hasIoControlList_;
-  bool hasInternalUnit_, hasNumberUnit_, hasStarUnit_;
-  bool hasCharFormat_, hasLabelFormat_, hasStarFormat_;
-  bool hasKnownAccess_, hasAccessDirect_, hasAccessStream_;
-  bool hasAdvanceYes_;
-  bool hasAsynchronousYes_;
-  bool hasKnownStatus_, hasStatusNew_, hasStatusReplace_, hasStatusScratch_;
-  bool hasDataList_;
+  common::EnumSet<Flag, Flag_enumSize> flag_;
```

Comment:
`flags_` or `flagSet_` would be a better name for this.

Reply:
Pending further discussion, I'll switch to flags_.

Reply:
Done